### PR TITLE
test: Add RegionType to test region helper

### PR DESCRIPTION
**Pull request by ykim-1**
## 📝 Description

Adding option to choose site type when choosing test region. (E.g. "core" or "distributed" for edge regions)

Also fixing broken E2E PR workflow with this PR

## ✔️ How to Test

Forked test run - https://github.com/ykim-1/terraform-provider-linode/actions/runs/10374018444/job/28720509675

## 📷 Preview

**If applicable, include a screenshot or code snippet of this change. Otherwise, please remove this section.**

### CODE REVIEW

File: linode/acceptance/util.go [hunk 1]
```diff
@@ -570,8 +570,12 @@ func ModifyProviderMeta(provider *schema.Provider, modifier ProviderMetaModifier
...
 }
 
-// GetRegionsWithCaps returns a list of regions that support the given capabilities.
-func GetRegionsWithCaps(capabilities []string, filters ...RegionFilterFunc) ([]string, error) {
+// GetRegionsWithCaps returns a list of region IDs that support the given capabilities
+// Parameters:
+// - capabilities: Required capabilities that the regions must support.
+// - siteType: The site type to filter by ("core" or "distributed").
+// - filters: Optional custom filters for additional criteria.
+func GetRegionsWithCaps(capabilities []string, regionType string, filters ...RegionFilterFunc) ([]string, error) {
```
Comment (lgarber-akamai) [inline linode/acceptance/util.go:578]: Would it make sense to also accept `any` as a regionType for situations where we aren't picky about core vs distributed?

Comment (ykim-1) [reply linode/acceptance/util.go:578]: Good call, I will add option to give `any` as an argument for `regionType`

**Comment by lgarber-akamai** [comment]
This change looks great! Will approve once the integration test results are back: https://github.com/linode/terraform-provider-linode/actions/runs/10374485461

**Comment by ykim-1** [comment]
> This change looks great! Will approve once the integration test results are back: https://github.com/linode/terraform-provider-linode/actions/runs/10374485461

So this E2E PR workflow was broken due to submodule not being checked out so I fixed it with this PR. I have a forked run [here ](https://github.com/ykim-1/terraform-provider-linode/actions/runs/10374018444/job/28720509675), sorry for the inconvenience. Failures are not related to this PR

**Comment by lgarber-akamai** [comment]
> > This change looks great! Will approve once the integration test results are back: https://github.com/linode/terraform-provider-linode/actions/runs/10374485461
> 
> So this E2E PR workflow was broken due to submodule not being checked out so I fixed it with this PR. I have a forked run [here ](https://github.com/ykim-1/terraform-provider-linode/actions/runs/10374018444/job/28720509675), sorry for the inconvenience. Failures are not related to this PR

Thanks for the super quick fix!

File: .github/workflows/integration_tests_pr.yml [hunk 2]
```diff
@@ -49,6 +50,8 @@ jobs:
         uses: actions/checkout@v4
         with:
           ref: ${{ inputs.sha }}
+          fetch-depth: 0
+          submodules: 'recursive'
```
Comment (lgarber-akamai) [inline .github/workflows/integration_tests_pr.yml:54]: Thanks for fixing this!

**Review by lgarber-akamai — APPROVED** [review]
Looks great after CI is fixed, nice work!

**Review by zliang-akamai — APPROVED** [review]
Looks good to me!

**Pull request merged**